File: intranet-angular.client/src/response/noticiaResponse.ts
import { CategoriaResponse } from "./categoriaResponse";
import { MidiaNoticiaResponse } from "./midiaNoticiaResponse";

export interface NoticiaResponse {
  id: number;
  titulo: string;
  conteudo: string;
  descricao: string;
  isTrendingTop: boolean;
  midiaNoticia: MidiaNoticiaResponse[];
  autorId?: number;
  autor: string;
  categoria: CategoriaResponse[];
  dataPublicacao: Date;
  categoriaNomes?: string;
}

File: intranet-angular.Server/Response/NoticiaResponse.cs
namespace intranet_angular.Server.Response
{
    public class NoticiaResponse
    {
        public int Id { get; set; }
        public string Titulo { get; set; }
        public string Descricao { get; set; }
        public string Conteudo { get; set; }
        public bool IsTrendingTop { get; set; }
        public DateTime DataPublicacao { get; set; }
        public int? AutorId { get; set; }
        public List<CategoriaResponse> Categoria { get; set; }
        public List<string> MidiaUrl { get; set; }
    }
}
